aws/sdk/integration-tests/s3/tests/streaming-response.rs

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::{Credentials, Endpoint, Region};
+use bytes::BytesMut;
+use std::net::SocketAddr;
+use std::time::Duration;
+use tracing::debug;
+
+// test will hang forever with the default (single-threaded) test executor
+#[tokio::test(flavor = "multi_thread")]
+#[should_panic(
+    expected = "error reading a body from connection: end of file before message length reached"
+)]
+async fn test_streaming_response_fails_when_eof_comes_before_content_length_reached() {
+    let addr = SocketAddr::from(([0, 0, 0, 0], 3000));
```

Comment:
I'd suggest using `0` as port here: the OS will assign a random free port and we don't risk to have test spuriously failing due to a port conflict (e.g. something else is already listening on 3000 and the `bind` call fails).

Reply:
Thanks for the suggestion. I updated the test.